### ISSUE TITLE
Unable to use Moq to unit test IMonitorClient

### ISSUE THREAD

**Issue by jeroenwo** [Azure/azure-sdk-for-net#3994]
I would like to use Moq as a mocking framework for unit testing code that uses the `IMonitorClient`. However, since the `ListAsync` method on the `IMetricsOperations` property is not part of the interface but an extension method in `MetricsOperationsExtensions`, I cannot fake a certain result with which I can test my code.

How do you test code which implements the `ListAsync` method?

Example test:
```csharp
[TestMethod]
public async Task Retriever_ServerFarmResource_ReturnsInstanceMetrics()
{
    var fixture = new Fixture();
    var monitorClientMock = new Mock<IMonitorClient>();
    var filterResponse = fixture.Create<Response>();

    // Setup the filter result.
    monitorClientMock
        .Setup(x => x.Metrics.ListAsync(It.IsAny<string>(), It.IsAny<ODataQuery<MetadataValue>>(), It.IsAny<string>(), It.IsAny<TimeSpan>(), It.IsAny<string>(), It.IsAny<string>(), ResultType.Metadata, CancellationToken.None))
        .ReturnsAsync(filterResponse);

    // Continue test using Mock.
}
```
This results in:
```
System.NotSupportedException: Invalid setup on an extension method: x => x.Metrics.ListAsync(It.IsAny<String>(), It.IsAny<ODataQuery\'1>(), It.IsAny<String>(), (Nullable'1)It.IsAny<TimeSpan>(), It.IsAny<String>(), It.IsAny<String>(), (Nullable'1)ResultType.Metadata, CancellationToken.None)
```
Issue is similar to #3968.

**Comment by dsgouda** [comment]
@jeroenwo The `ListAsync` method is basically a wrapper around the `ListWithHttpMessagesAsync` method [here](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Monitor/Management.Monitor/Generated/IMetricsOperations.cs#L83). You can wrap the function call just like the extension method does [here](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Monitor/Management.Monitor/Generated/MetricsOperationsExtensions.cs#L119)
If this still does not unblock you, please reopen the issue with related details.
